sql/core/src/main/scala/org/apache/spark/sql/execution/command/functions.scala

```diff
@@ -236,6 +236,45 @@ case class ShowFunctionsCommand(
   }
 }
 
+
+/**
+ * A command for users to refresh the persistent function.
+ * The syntax of using this command in SQL is:
+ * {{{
+ *    REFRESH FUNCTION functionName
+ * }}}
+ */
+case class RefreshFunctionCommand(
+    databaseName: Option[String],
+    functionName: String)
+  extends RunnableCommand {
+
+  override def run(sparkSession: SparkSession): Seq[Row] = {
+    val catalog = sparkSession.sessionState.catalog
+    if (FunctionRegistry.builtin.functionExists(FunctionIdentifier(functionName))) {
```

Comment:
We still can create persistent function with the same name as the built-in function. For example, 

```SQL
CREATE FUNCTION rand AS 'org.apache.spark.sql.catalyst.expressions.Abs'
DESC function default.rand
```

I think we should still allow this case.

Reply:
It seems no meaning to refresh a persistent function whose name is same as a built-in function.

Yes, we can create a persistent function with the same name as the built-in function, but just create in metastore. The actual function we used is the built-in function. The reason is built-in functions are pre-cached in registry and we lookup cached function first.

e.g., `CREATE FUNCTION rand AS 'xxx'`, `DESC FUNCTION rand` will always return `Class: org.apache.spark.sql.catalyst.expressions.Rand`.

BTW, maybe it's the reason why we create function and load it lazy that just be a Hive client, otherwise we can't create such function like `rand`,`md5` in metastore. @cloud-fan

Reply:
how about
```
CREATE FUNCTION rand AS 'xxx';
DESC FUNCTION default.rand;
```

I think this is similar to table and temp views. Spark will try to look up temp view first, so if the name conflicts, temp view is preferred. But users can still use a qualified table name to read the table explicitly.

Reply:
You are right.

Missed qualified name case, I will fix this in followup.